google/cloud/firestore_v1/aggregation.py

```diff
@@ -154,14 +184,20 @@ def _make_stream(
 
             if response is None:  # EOI
                 break
+
+            if response.explain_metrics:
+                yield None, response.explain_metrics
```

Comment:
no need to yield here: you can yield them both together a couple lines later. That way, only the second value needs to be optional

```
maybe_metrics = response.explain_metrics or None
result = _query_response_to_result(response)
yield result, maybe_metrics
```

Reply:
I like the idea of returning the value instead! This way we can avoid calling next() internally too.